pandas/_testing.py

```diff
@@ -667,6 +667,7 @@ def assert_index_equal(
     check_less_precise: Union[bool, int] = no_default,
     check_exact: bool = True,
     check_categorical: bool = True,
+    check_order: bool = True,
     rtol: float = 1.0e-5,
```

Comment:
Probably it is necessary to have the new parameter added to the very end of this function signature in order not to break any user's code, which for some reason use positional arguments.

Reply:
Thanks for the feedback.

I debated that and it looked like the ```rtol``` and ```atol``` params had been added before the (older) ```obj``` param, so I decided to put ```check_order``` with the other check params.  That said, ```obj``` is not something that most user code would pass so perhaps the comparrison doesn't hold.

Happy to move the parameter.

I also just noticed that I forgot to add a ```versionadded``` tag (will do).